File: Leetcode/TwoPointers/minimunCommonValue.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
const getCommon = function(nums1, nums2) {
    let ans = 0;
    let sp = 0;
    let p = 0;
    const largestArray = nums1.length < nums2.length ? nums2.length : nums1.length;
    for (let i = 0; i < largestArray; i++) {
        const k = nums1[sp];
        console.log("This is K", k);
        const z = nums2[p];
        console.log("This is Z", z);
        if (k === z) {
            ans = k;
            console.log("This is also ANS", ans);
            return ans;
        }
        if (k < z) {
            sp++;
        } else {
            p++;
        }
    }
    ans = - 1;
    console.log("This is ANS",ans);
    return ans;
};

const nums1 = [10,2,3,4];
const nums2 = [2,3,4,5,10];

getCommon(nums1, nums2);

File: Leetcode/TwoPointers/MinimunCommonValue.java
package Leetcode.TwoPointers;

public class MinimunCommonValue {
    public static void main(String[] args) {
        int[] A = {0,4,5,6,8,10,11};
        int[] N = {1,2,3,4,6,15};
        System.out.println("The minimun common number is: ");
        int ans = minimunCommonValue(A, N);
        System.out.println(ans);

    }

    private static int minimunCommonValue(int[] A, int[] N) {
        int ans = 0;
        int sp = 0;
        int p = 0;
        int largestArray = A.length < N.length ? N.length : A.length;

        for (int i = 0; i < largestArray; i++) {
            int k = A[sp];
            int z = N[p];
            if (k == z) {
                ans = z;
                return z;
            }
            if (k < z) {
                sp++;
            } else {
                p++;
            }
        }
        ans = -1;
        return ans;
    }
}

